tests/ui/derived_hash_with_manual_eq.rs

```diff
@@ -34,23 +34,4 @@ impl std::hash::Hash for Bah {
     fn hash<H: std::hash::Hasher>(&self, _: &mut H) {}
 }
 
-#[derive(PartialEq)]
-struct Foo2;
-
-trait Hash {}
-
-// We don't want to lint on user-defined traits called `Hash`
-impl Hash for Foo2 {}
-
-mod use_hash {
-    use std::hash::{Hash, Hasher};
-
-    #[derive(PartialEq)]
-    struct Foo3;
-
-    impl Hash for Foo3 {
-        fn hash<H: std::hash::Hasher>(&self, _: &mut H) {}
-    }
```

Comment:
I think it would be better to keep this test code and add a comment explaining, why this is okay. Something simple like:

```
/// Implementing `Hash` with a derived `PartialEq` is fine. See #2627
```

Reply:
Ah, okay I misunderstood this part. Thank you :)